polaris-react/playground/Typography.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {ReactNode} from 'react';
+
+import {classNames, variationName} from '../src/utilities/css';
+
+import styles from './Typography.scss';
+
+export type ElementType =
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'h6'
+  | 'a'
+  | 'p'
+  | 'span'
+  | 'div';
+
+type FontWeight = 'regular' | 'medium' | 'semibold' | 'bold';
+
+interface BaseTypographyProps {
+  as: ElementType;
+  children: ReactNode;
+  fontWeight?: FontWeight;
+  noWrap?: boolean;
+  size?: string;
+  variant?: string;
+  // fontSize?: string;
+}
+
+interface TypographyProps extends BaseTypographyProps {
+  variant: string;
+  size: string;
+}
+
+export const Typography = ({
```

Comment:
Suggest updating the component name to `Text` (e.g. [React Native](https://reactnative.dev/docs/text), [React Ink](https://github.com/vadimdemedes/ink#text)). I have a hunch the additional characters will influence folks opinions of the component as opposed to comparing the styling APIs.